test/ci/verify-templates.test.js

```diff
@@ -0,0 +1,137 @@
+              (file) =>
+              file.isDirectory() &&
+                  !file.name.startsWith('.') &&
+                  !file.name.startsWith('_') &&
+                  !excludedPaths.includes(file.name)
+          )
+          .map((dir) => dir.name);
+
+    describe('CI template verification', () => {
+        describe.each(templates)('the "%s" function template', (template) => {
+            it('should have a populated functions directory', (done) => {
+                const functionsDir = path.join(projectRoot, template,
+                                               'functions');
+
+                fs.readdir(functionsDir, (err, files) => {
+                    expect(err).toBeFalsy();
+                    expect(files.length).toBeGreaterThan(0);
+                    done();
+                });
+            });
+
+            it('should have a populated tests directory', (done) => {
+                const functionsDir = path.join(projectRoot, template, 'tests');
+
+                fs.readdir(functionsDir, (err, files) => {
+                    expect(err).toBeFalsy();
+                    expect(files.length).toBeGreaterThan(0);
+                    done();
+                });
+            });
+
+            describe('its assets/index.html file', () => {
+                const indexFile = path.join(projectRoot, template,
+                                            'assets', 'index.html');
+
+                it('should exist', (done) => {
+                    fs.access(indexFile, fs.constants.F_OK, (err) => {
+                        expect(err).toBeFalsy();
+                        done();
+                    });
+                });
+
+                it('should contain <!-- APP_INFO_V2 --> (for new-style index pages)', (done) => {
+                    fs.readFile(indexFile, (err, data) => {
+                        expect(err).toBeFalsy();
+                        if(data.includes('ce-paste-theme.css')) {
+                            expect(data.includes('<!-- APP_INFO_V2 -->'));
+                        }
+                        done();
+                    });
+                });
+            });
+
+            describe('its .env file', () => {
+                const envFile = path.join(projectRoot, template, '.env');
+
+                it('should exist', (done) => {
+                    fs.access(envFile, fs.constants.F_OK, (err) => {
+                        expect(err).toBeFalsy();
+                        done();
+                    });
+                });
+
+                it('should be parseable', async () => {
+                    const result = parser.parseFile(envFile);
+
+                    await expect(result).resolves.toBeTruthy();
+                });
+
+                it('should have a description for each variable', async () => {
+                    const result = await parser.parseFile(envFile);
+
+                    result.variables.forEach((v) => {
+                        if(!v.description) {
```

Comment:
any reason not to use the expect framework here? something like 
```
result.variables.forEach((v) => expect(v.description).toBeTruthy())
```

Reply:
I did it this way to get Jest to print a custom error message that calls out the specific variable name that doesn't have a description, since the default assert error is a bit confusing without it. Otherwise, there's unfortunately [no way to get `expect` to report a custom error message](https://github.com/facebook/jest/issues/3293) out of the box. I can pull in one of the Jest plugins that adds this capability if you'd prefer a more Jest-like solution; I didn't know if this was a pattern we wanted to use elsewhere, so I hesitated initially to pull in a `devDependency` for this.